sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/TracerProvider.java

```diff
@@ -289,10 +281,18 @@ public void addEvent(String name, Map<String, Object> attributes, OffsetDateTime
 
     private void end(int statusCode, Throwable throwable, Context context) {
         if (throwable != null) {
-            tracer.setAttribute(TracerProvider.ERROR_MSG, throwable.getMessage(), context);
-            tracer.setAttribute(TracerProvider.ERROR_TYPE, throwable.getClass().getName(), context);
+            if (statusCode == HttpConstants.StatusCodes.NOTFOUND) {
+                tracer.setAttribute(TracerProvider.ERROR_MSG, "Not found exception", context);
+                tracer.setAttribute(TracerProvider.ERROR_TYPE, throwable.getClass().getName(), context);
+                tracer.end(statusCode, null, context);
+            } else {
+                tracer.setAttribute(TracerProvider.ERROR_MSG, throwable.getMessage(), context);
```

Comment:
BTW, why are we adding those? 
- `tracer.end` will do `span.recordException(t)`

https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/implementation/HttpTraceUtil.java#L53

which internally record common exception attributes 
https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/trace/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java#L406-L409

So we can remove all ERROR_TYPE/ERROR_MSG attributes logic?

Reply:
If we remove ERROR_TYPE/ERROR_MSG then we wont be seeing below attributes which was the initial requirement from @trask 
![image](https://user-images.githubusercontent.com/43345335/134070653-838e0896-2bd8-4cd1-8c67-21bd4b8e610b.png)

Reply:
Can you please add screenshot for non-404 case? There will be standard otel attributes recorded for exception (`exception.type` and `exceptions.message`)?

@trask  I believe you wanted to have otel semantic exception attributes attached, they have changes and now we are populating new ones + old ones. Are you ok with cosmos removing `error.msg` and `error.type`? Do you believe we use them anywhere?

Reply:
yes, I think we should remove those, I had similar thought a few weeks ago but got distracted from opening PR 😓: https://github.com/trask/azure-sdk-for-java/commit/af44c693a9e4bbf1013857ec34724a3412216d0f

Reply:
@trask  Removed  error.msg and error.type.
@lmolkova  please see the screen shot when it is not 404/0 (item or collection not found). We cannot differentiate between item and collection not found without parsing cosmos exception diagnostics and it will be error prone in future.
However if the customer logging the exception they can still see the full diagnostics and differentiate

![image](https://user-images.githubusercontent.com/43345335/134208994-cd457bef-e1f6-42f9-8431-1ed82741306e.png)